CMakeLists.txt

```diff
@@ -53,6 +53,10 @@ if (BUILD_DEPS)
                 NOT (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l))
                 set(DISABLE_PERL ON)
             endif()
+            if(CMAKE_C_COMPILER_ID MATCHES "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_LESS "5.0")
+                set(DISABLE_PERL OFF CACHE BOOL "Build with Perl to avoid using pre-compiled binary with AVX512")
+                set(MY_ASSEMBLER_IS_TOO_OLD_FOR_512AVX ON CACHE BOOL "Disable AVX512 on old GCC that not supports it")
+            endif()
```

Comment:
Maybe add a comment about why we are doing this.

Reply:
I think it's self-explained -> "GNU < 5.0 is too old for AVX512"